src/UWP/Services/KeyDerivationService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Bit.App.Abstractions;
+using Org.BouncyCastle.Crypto.Digests;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Security.Cryptography;
+using Windows.Security.Cryptography.Core;
+using Windows.Storage.Streams;
+
+namespace Bit.UWP.Services
+{
+    public class KeyDerivationService : IKeyDerivationService
+    {
+        private const int KeyLength = 256; // 32 bytes
+
+        //todo review this
+        public byte[] DeriveKey(byte[] password, byte[] salt, uint rounds)
```

Comment:
Seems fine since you are using the same BC implementation as Android. I know .NET Core has a native implementation for PBKDF2 so maybe we can switch to that when we move to .NET standard.

Reply:
I didn't know about native implementation of PBKDF2, I added it now